### PR TITLE
[SMALLFIX] Adding missing jets3t dependencies into the S3 UFS jar.

### PR DESCRIPTION
**Pull request by jsimsa**
@apc999

### CODE REVIEW

File: underfs/s3/pom.xml [hunk 1]
```diff
@@ -37,6 +37,30 @@
       <version>0.8.1</version>
     </dependency>
 
+    <!-- The following are jets3t dependencies
+     (see http://mvnrepository.com/artifact/net.java.dev.jets3t/jets3t/0.8.1).
+     We need to explicitly include them, because jets3t is shaded. -->
```
Comment (calvinjia) [inline underfs/s3/pom.xml]: Is there an option to do this automatically? (`promoteTransitiveDependencies`?)

Comment (jsimsa) [inline underfs/s3/pom.xml]: I tried that and it fails with:

```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-shade-plugin:2.2:shade (shade-jets3t) on project tachyon-underfs-s3: Error creating shaded jar: Could not resolve dependencies for project org.tachyonproject:tachyon-underfs-s3:jar:0.9.0-SNAPSHOT: The following artifacts could not be resolved: com.sun.jdmk:jmxtools:jar:1.2.1, com.sun.jmx:jmxri:jar:1.2.1: Could not transfer artifact com.sun.jdmk:jmxtools:jar:1.2.1 from/to java.net (https://maven-repository.dev.java.net/nonav/repository): Cannot access https://maven-repository.dev.java.net/nonav/repository with type legacy using the available connector factories: BasicRepositoryConnectorFactory: Cannot access https://maven-repository.dev.java.net/nonav/repository with type legacy using the available layout factories: Maven2RepositoryLayoutFactory: Unsupported repository layout legacy -> [Help 1]
```

**Comment by apc999** [comment]
+1 LGTM.

**Comment by jsimsa** [comment]
@calvinjia does this look go to you?

**Comment by aaudiber** [comment]
LGTM

File: underfs/s3/pom.xml (hunk 1 above)
Comment (calvinjia) [inline underfs/s3/pom.xml]: Could we leave a TODO to investigate this later?

**Comment by calvinjia** [comment]
LGTM

File: underfs/s3/pom.xml (hunk 1 above)
Comment (jsimsa) [inline underfs/s3/pom.xml]: Sorry I missed your comment before merging this in, I will open a new PR for it.

Comment (jsimsa) [inline underfs/s3/pom.xml]: See https://github.com/amplab/tachyon/pull/2488

**Pull request merged**